console/src/channel/CreateCalculated.tsx

```diff
@@ -0,0 +1,311 @@
+    message: "Index channel must have data type TIMESTAMP",
+    path: ["dataType"],
+  })
+  .refine((v) => v.isIndex || v.index !== 0 || v.virtual, {
+    message: "Data channel must have an index",
+    path: ["index"],
+  });
+
+type FormValues = z.infer<typeof schema>;
+
+export const CREATE_CALCULATED_LAYOUT_TYPE = "createCalculatedChannel";
+
+const SAVE_TRIGGER: Triggers.Trigger = ["Control", "Enter"];
+
+export const createCalculatedLayout: Layout.State = {
+  key: CREATE_CALCULATED_LAYOUT_TYPE,
+  type: CREATE_CALCULATED_LAYOUT_TYPE,
+  windowKey: MAIN_WINDOW,
+  name: "Channel.Create.Calculated",
+  icon: "Channel",
+  location: "modal",
+  tab: {
+    closable: true,
+    editable: false,
+  },
+  window: {
+    resizable: false,
+    size: { height: 600, width: 1000 },
+    navTop: true,
+    showTitle: true,
+  },
+};
+
+const ZERO_FORM_VALUES: FormValues = {
+  key: 0,
+  name: "",
+  index: 0,
+  dataType: "float32",
+  isIndex: false,
+  leaseholder: 0,
+  virtual: true,  // Set to true by default
+  rate: Rate.hz(0),
+  internal: false,
+  expression: "np.array([])",
+  requires: [],
+};
+
+export const CreateCalculatedModal: Layout.Renderer = ({ layoutKey, onClose }) => {
+  const client = Synnax.use();
+  const args = Layout.useSelectArgs<CalculatedChannelArgs>(layoutKey) ?? defaultArgs
+  const res = useQuery<FormValues>({
+    queryKey: [args.channelKey, client?.key],
+    queryFn: async () => {
+      if (args.channelKey == null) return deep.copy(ZERO_FORM_VALUES);
+      if (client == null) throw new Error("Client not available");
+      const ch = await client.channels.retrieve(args.channelKey);
+      return { ...ch, dataType: ch.dataType.toString() };
+    },
+  });
+
+  if (res.isLoading) return <Text.Text level="p">Loading...</Text.Text>;
+  if (res.isError)
+    return (
+      <Align.Space direction="y" grow style={{ height: "100%" }}>
+        <Status.Text.Centered variant="error">{res.error.message}</Status.Text.Centered>
+      </Align.Space>
+    );
+  if (res.isSuccess) return <Internal onClose={onClose} initialValues={res.data} />;
+  return null;
+};
+
+interface InternalProps extends Pick<RendererProps, "onClose"> {
+  initialValues: FormValues;
+}
+
+const Internal = ({ onClose, initialValues }: InternalProps): ReactElement => {
+  const client = Synnax.use();
+
+  const methods = Form.use<typeof schema>({ schema, values: initialValues });
+
+  const addStatus = Status.useAggregator();
+
+  const [createMore, setCreateMore] = useState(false);
+  const { mutate, isPending } = useMutation({
+    mutationFn: async (createMore: boolean) => {
+     if(!methods.validate() || client == null) return;
+      const d = methods.value();
+      await client.channels.create(d);
+      if (!createMore) onClose();
+      else methods.reset(deep.copy(ZERO_FORM_VALUES));
+    },
+    onError: (error: Error) => {
+      addStatus({
+        variant: "error",
+        message: "Error creating channel",
```

Comment:
"Error creating calculated channel". Might also be good to include the name of the channel?